pkg/netpol/connlist/exposed_pods.go

```diff
@@ -0,0 +1,28 @@
+package connlist
+
+import "github.com/np-guard/netpol-analyzer/pkg/netpol/internal/common"
+
+// ExposedPeer captures potential ingress and egress connections data for an exposed Peer
+type ExposedPeer interface {
+	// ExposedPeer is a peer for which the analysis found some potential exposure info
+	ExposedPeer() Peer
+	// IngressExposure is a list of the potential Ingress connections to the ExposedPeer
+	IngressExposure() []XgressExposureData
+	// EgressExposure is a list of the potential Egress connections from the ExposedPeer
+	EgressExposure() []XgressExposureData
+}
+
+// XgressExposureData contains data of potential connectivity for an existing peer to/from a representative peer
+type XgressExposureData interface {
+	// IsProtectedByNetpols indicates if the exposed peer is protected by any netpol on Ingress/Egress
+	// if a peer is not protected by xgress netpols, it will be exposed to entire cluster with all allowed connections
+	IsProtectedByNetpols() bool
+	// IsExposedToEntireCluster indicates if the peer is exposed to all namespaces in the cluster for the relevant direction
+	IsExposedToEntireCluster() bool
+	// NamespaceLabels are matchLabels of potential namespaces which the peer might be exposed to
+	NamespaceLabels() map[string]string
+	// PodLabels are matchLabels of potential pods which the peer might be exposed to
+	PodLabels() map[string]string
+	// PotentialConnectivity the potential connectivity of the exposure
+	PotentialConnectivity() common.AllowedConnectivity
```

Comment:
is it possible to infer from the possible values of objects that implement this interface, the mapped node type in the planned output? 

what about the case of a specific namespace with specific pod labels (ns inferred from netpol)? how is it represented by such an object?
what about the case of specific pod lables with all namspaces? 

```
the new nodes types :

    any-namespace
    any namespace with labels
    any pod with labels in the netpol's namespace
    any pod with labels in any-namespace
    any pod with labels in any namespace with labels
```

Reply:
- for the specific namespace i think we may use the name selector of the namespace.
- for pods in any-namespace maybe if the namespaceLabels is empty (or with flag IsExposedToEntireCluster() true)

Reply:
ok, should also update documentation to explain this mapping, so that one can interpret the objects correctly

Reply:
> for the specific namespace i think we may use the name selector of the namespace.
I think this way we wouldn't know whether we can specify the namespace by name or by labels, but we can handle this at a later stage (ns by labels is originated from selector, ns by name is originated by netpol ns ... )